### ISSUE TITLE
[Sonos] Request: Favorites should be chosen by name in Routines

### ISSUE THREAD

**Issue by TheMegamind** [SmartThingsCommunity/SmartThingsEdgeDrivers#618]
In the driver's current form, when a user attempts to load a favorite via a routine, they are asked to make the following entry:

![2023-03-10 13 08 37](https://user-images.githubusercontent.com/12296628/224429211-46889b45-9322-449b-8486-348dbc04bc8d.jpg)

What is not clear, and causing issues among those users, is that the form is looking for a favoriteId, not a favorite name as most users expect. Even if this were clear, however, only a tiny minority of users are aware that favorites have ID numbers, and even fewer have any idea how to reference them. This renders this capability effectively useless for most. 

This form should instead accept favorite names, either through direct text entry or via a dropdown similar to that available in the device presentation (although I'm unsure if this is even possible).  If it is,  it is further suggested that the favorites be alphabetized within the dropdown to make them easier to find and select. 

It seems likely something this is already on the roadmap, but wanted to raise the issue in case it was inadvertently overlooked.

cc: @dljsjr

**Comment by dljsjr** [comment]
Hey @TheMegamind, thanks for the report! Sorry for the delayed response, I had to do some investigating on this ticket first.

This is actually a platform bug upstream of the edge driver, so I'm going to close this GitHub issue since we want to keep these limited to bugs that are in the edge driver code itself.

But I did want you to know that it was ingested in to our internal bug tracker and that we are aware and working on it!
